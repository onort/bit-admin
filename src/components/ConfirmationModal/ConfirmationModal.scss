@import "../../styles/main.scss";

.container {
  position: relative;
  margin: 20vh auto 0 !important;
  max-width: 80% !important;
  width: px-to-rem(480px) !important;
  border-radius: px-to-rem(4px) !important;
  background-color: $white;
  padding: $spacer-xlg $spacer-lg $spacer-md !important;
}

.content {
  padding: $spacer-lg 0 0;
}

.actions {
  display: flex;
  justify-content: flex-end;
  margin-top: $spacer-xlg;
  padding-top: $spacer-sm;
  border-top: 1px solid $light-border;
}

.confirm {
  background-color: $green;
  color: $white;

  &:hover {
    background-color: lighten($green, 10);
  }
}

.cancel {
  background-color: $danger;
  color: $white;
  margin-left: $spacer-md;

  &:hover {
    background-color: lighten($danger, 10);
  }
}

.close {
  position: absolute;
  top: $spacer-md;
  right: $spacer-lg;
  cursor: pointer;
  stroke-width: 1px;
  color: $text-color-muted;
}
