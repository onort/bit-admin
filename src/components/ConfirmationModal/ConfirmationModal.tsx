import React from "react"
import { MdClose as CloseIcon } from "react-icons/md"

import styles from "./ConfirmationModal.scss"
import { Button, Modal, Paper } from ".."

interface Props {
  cancelText?: string
  confirmText?: string
  text: string
  onCancel: (e: React.MouseEvent) => void
  onClose?: (e: React.MouseEvent) => void
  onConfirm: (e: React.MouseEvent) => void
}

const ConfirmationModal: React.SFC<Props> = props => {
  const { cancelText, confirmText, onCancel, onClose, onConfirm, text } = props
  return (
    <Modal>
      <Paper elevation={4} className={styles.container}>
        {onClose && <CloseIcon className={styles.close} onClick={onClose} />}
        <div className={styles.content}>
          <p className={styles.text}>{text}</p>
        </div>
        <div className={styles.actions}>
          <Button
            className={styles.confirm}
            text={confirmText}
            onClick={onConfirm}
          />
          <Button
            className={styles.cancel}
            text={cancelText}
            onClick={onCancel}
          />
        </div>
      </Paper>
    </Modal>
  )
}

ConfirmationModal.defaultProps = {
  cancelText: "Cancel",
  confirmText: "Accept"
}

export default ConfirmationModal
