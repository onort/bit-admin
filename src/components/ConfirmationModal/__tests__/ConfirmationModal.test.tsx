import React from "react"
import { shallow } from "enzyme"

import ConfirmationModal from "../"

const fakeCancel = jest.fn()
const fakeClose = jest.fn()
const fakeConfirm = jest.fn()
const testText = "Test text."

describe("<ConfirmationModal />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <ConfirmationModal
        text={testText}
        onCancel={fakeCancel}
        onConfirm={fakeConfirm}
      />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have text rendered", () => {
    const wrapper = shallow(
      <ConfirmationModal
        text={testText}
        onCancel={fakeCancel}
        onConfirm={fakeConfirm}
      />
    )
    expect(wrapper.find(".text").text()).toBe(testText)
  })

  it("should render correctly if a cancelText prop is passed", () => {
    const wrapper = shallow(
      <ConfirmationModal
        cancelText={testText}
        text={testText}
        onCancel={fakeCancel}
        onConfirm={fakeConfirm}
      />
    )
    expect(wrapper.find(".cancel").prop("text")).toBe(testText)
  })

  it("should render correctly if a confirmText prop is passed", () => {
    const wrapper = shallow(
      <ConfirmationModal
        confirmText={testText}
        text={testText}
        onCancel={fakeCancel}
        onConfirm={fakeConfirm}
      />
    )
    expect(wrapper.find(".confirm").prop("text")).toBe(testText)
  })

  it("should render correctly if a onClose prop is passed", () => {
    const wrapper = shallow(
      <ConfirmationModal
        text={testText}
        onCancel={fakeCancel}
        onClose={fakeClose}
        onConfirm={fakeConfirm}
      />
    )
    expect(wrapper.find(".close").exists()).toBe(true)
  })

  it("should fire onClose function if closeIcon is clicked", () => {
    const closeFn = jest.fn()
    const wrapper = shallow(
      <ConfirmationModal
        text={testText}
        onCancel={fakeCancel}
        onClose={closeFn}
        onConfirm={fakeConfirm}
      />
    )
    wrapper.find(".close").simulate("click")
    expect(closeFn).toBeCalledTimes(1)
  })

  it("should fire onCancel function if cancel button is clicked", () => {
    const cancelFn = jest.fn()
    const wrapper = shallow(
      <ConfirmationModal
        text={testText}
        onCancel={cancelFn}
        onConfirm={fakeConfirm}
      />
    )
    wrapper.find(".cancel").simulate("click")
    expect(cancelFn).toBeCalledTimes(1)
  })

  it("should fire onConfirm function if confirm button is clicked", () => {
    const confirmFn = jest.fn()
    const wrapper = shallow(
      <ConfirmationModal
        text={testText}
        onCancel={fakeCancel}
        onConfirm={confirmFn}
      />
    )
    wrapper.find(".confirm").simulate("click")
    expect(confirmFn).toBeCalledTimes(1)
  })
})
