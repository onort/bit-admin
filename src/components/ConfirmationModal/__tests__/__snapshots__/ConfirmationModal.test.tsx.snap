// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> should match snapshot 1`] = `
<Modal>
  <Paper
    className="container"
    elevation={4}
  >
    <div
      className="content"
    >
      <p
        className="text"
      >
        Test text.
      </p>
    </div>
    <div
      className="actions"
    >
      <Button
        className="confirm"
        onClick={[MockFunction]}
        text="Accept"
      />
      <Button
        className="cancel"
        onClick={[MockFunction]}
        text="Cancel"
      />
    </div>
  </Paper>
</Modal>
`;

exports[`<ConfirmationModal /> should match snapshot 2`] = `"<div class=\\"container\\"><div class=\\"container elevation4 container\\"><div class=\\"content\\"><p class=\\"text\\">Test text.</p></div><div class=\\"actions\\"><button class=\\"button confirm\\"><span class=\\"text\\">Accept</span></button><button class=\\"button cancel\\"><span class=\\"text\\">Cancel</span></button></div></div></div>"`;
